lib/cable_ready/channels.rb

```diff
@@ -4,12 +4,24 @@
 
 module CableReady
   class Channels
+    include Singleton
+    attr_accessor :tools
+
+    def self.configure
+      yield CableReady::Channels.instance if block_given?
+    end
+
     def initialize
       @channels = {}
+      @tools = [:add_css_class, :dispatch_event, :inner_html, :insert_adjacent_html, :insert_adjacent_text, :morph, :outer_html, :remove, :remove_attribute, :remove_css_class, :set_attribute, :set_cookie, :set_dataset_property, :set_property, :set_style, :set_styles, :set_value]
+    end
+
+    def add_tool(tool)
+      @tools << tool.to_sym
     end
```

Comment:
Would it make sense to allow a block here that provides the body of the tool's method?

Something like

``` ruby
def add_tool(tool, &implementation)
  @tools[tool] = implementation || ->(options = {}) { add_operation(tool, options) }
end
```

And then

``` ruby
def initialize(name, tools)
  ...

  tools.each do |tool, implementation|
    define_singleton_method(tool, &implementation)
  end
end
```

I don't really have a specific use case in mind, and this might be making internal APIs more public than they need to be, but if the goal is to reduce the necessity for monkey patching that this could come in handy if someone needs to do something more custom in their operation.

WDYT?

Reply:
I *love* this idea. I'm just going to borrow it and pretend like it was my idea, okay? ;)

Seriously - this is great.